### ISSUE TITLE
Unable to create a card because it's a duplicate (not no ANKI_NOTE_ID is inserted)

### ISSUE THREAD

**Issue by dmitrym0** [eyeinsky/org-anki#45]
Hello folks,

I created a small helper function to ankify a subtree of cards. I noticed after running it that some of the headings did not acquired the `ANKI_NOTE_ID` property but they did show up in the Anki database. Obviously when I tried reinserting them they failed with the duplicate error. 

Has anyone else observed this? I'm basically running `org-anki-sync-entry` on a subtree. Nothing complicated.

**Comment by hwiorn** [comment]
> Obviously when I tried reinserting them they failed with the duplicate error.

Because [anki-connect](https://github.com/FooSoft/anki-connect) checks a first field for duplication. Simply, org-anki uses Basic card type which first fields is a headline by default. You need to make disable duplication on org-anki or set a different first field of each cards in your helper function.

**Comment by eyeinsky** [comment]
If you need to add an org subtree of entries to Anki then an easier way to add them would be to paste the entire thing into a temporary org buffer and add everything there and them move them back -- now each entry having an `ANKI_NOTE_ID` attribute. I.e it doesn't matter from which file the notes were added from.

**Comment by eyeinsky** [comment]
If notes are partially added to Anki then see if there is any info in the `*Messages*`. Closing this issue for now.

**Comment by kenhuang-io** [comment]
> I created a small helper function to ankify a subtree of cards. I noticed after running it that some of the headings did not acquired the `ANKI_NOTE_ID` property but they did show up in the Anki database.

I've seen this, I think it's a bug, but I'm not sure where it is.

**Comment by eyeinsky** [comment]
Does this happen with the `org-anki-sync-all` command and is there a way it
could be reproduced?--
Markus Läll

**Comment by kenhuang-io** [comment]
Hi Markus,

I think it's reproducible. I encountered this for a few times in the past and just yesterday, when I tried to sync a few hundreds of entries in a batch, using my own recipe of `w/org-anki-sync-subentries`, which was mentioned [here](https://github.com/eyeinsky/org-anki/issues/66#issuecomment-1596089022).

Sorry that I don't have too much time recently, but I will investigate it further in the future if I have the time.
